pyasn1/codec/ber/decoder.py

```diff
@@ -276,24 +135,27 @@ def valueDecoder(self, substrate, asn1Spec,
         if tagSet[0].tagFormat != tag.tagFormatSimple:
             raise error.PyAsn1Error('Simple tag format expected')
 
-        the_bytes = substrate.read(length)
-        if not the_bytes:
-            return self._createComponent(asn1Spec, tagSet, 0, **options)
+        for chunk in streaming.read(substrate, length, options):
+            if isinstance(chunk, SubstrateUnderrunError):
```

Comment:
I am still suspicious wrt to `isinstance` versus `is` in all these conditions. Would be interesting to replace one with the other and benchmark the difference...